torch/_higher_order_ops/utils.py

```diff
@@ -969,6 +972,36 @@ def call_op(op: Union[OpOverload, HopInstance], args, kwargs):
     return op(*bound_args, **bound_kwargs)
 
 
+def materialize_as_graph(
+    fn: Callable,
```

Comment:
This is a bare callable, please add some typing support here for return value and input params with ParamSpec?